conbench/entities/benchmark_result.py

```diff
@@ -1025,6 +1041,30 @@ def _guts(cinfo: TypeCommitInfoGitHub) -> Tuple[Commit, bool]:
     return commit
 
 
+def generate_history_fingerprint(
+    case_id: str, context_id: str, hardware_hash: str, repo_url: str
+) -> str:
+    """Generate a history_fingerprint from the relevant control variables."""
+    hash = hashlib.md5()
+
+    # The unique index on Case means that its primary key is a unique identifier for the
+    # specific benchmark name and case permutation (tags)
+    hash.update(case_id.encode("utf-8"))
+
+    # Similarly, Context has a unique index on its tags
+    hash.update(context_id.encode("utf-8"))
+
+    # The conventional field for checking whether two hardwares are compatible is
+    # "hash". This might change in the future:
+    # https://github.com/conbench/conbench/issues/1281
+    hash.update(hardware_hash.encode("utf-8"))
+
+    # Don't mix two repositories
+    hash.update(repo_url.encode("utf-8"))
+
+    return hash.hexdigest()
```

Comment:
Good!

Future readers / devs might wonder about the choice of MD5 -- it's fine, it's fast, and I think collisions are practically impossible (but that back-of-the-envelope calculation should at some point maybe be done).

Reply:
I copied this comment directly into a code comment :) thanks!